modules/runtime-common/src/main/scala/com/snowplowanalytics/snowplow/runtime/Webhook.scala

```diff
@@ -42,32 +47,61 @@ object Webhook {
     }
   }
 
-  private implicit def logger[F[_]: Sync]: Logger[F] = Slf4jLogger.getLogger[F]
+  def resource[F[_]: Async, SetupAlert: Show: Eq](
+    config: Config,
+    appInfo: AppInfo,
+    httpClient: Client[F],
+    appHealth: AppHealth[F, SetupAlert, ?]
+  ): Resource[F, Unit] =
+    stream(config, appInfo, httpClient, appHealth).compile.drain.background.void
 
-  def create[F[_]: Async, Alert: Show](
-    config: Option[Config],
+  def stream[F[_]: Async, SetupAlert: Show: Eq](
+    config: Config,
     appInfo: AppInfo,
-    httpClient: Client[F]
-  ): Webhook[F, Alert] = new Webhook[F, Alert] {
-
-    override def alert(message: Alert): F[Unit] =
-      config match {
-        case Some(webhookConfig) =>
-          val request = buildHttpRequest[F, Alert](webhookConfig, appInfo, message)
-          Logger[F].info(show"Sending alert to ${webhookConfig.endpoint} with details of the setup error...") *>
-            executeHttpRequest[F](webhookConfig, httpClient, request)
-        case None =>
-          Logger[F].debug(s"Webhook monitoring is not configured, skipping alert: $message")
+    httpClient: Client[F],
+    appHealth: AppHealth[F, SetupAlert, ?]
+  ): Stream[F, Nothing] =
+    appHealth.setupHealth.discrete.changes
+      .through(repeatPeriodically(config.heartbeat))
```

Comment:
About `repeatPeriodically`, I wonder if we could achieve the same using this:

```scala

    val lastValueHeartbeat = Stream.fixedDelay(config.heartbeat) *> Stream.eval(appHealth.setupHealth.get)
    val healthChanges = appHealth.setupHealth.discrete.changes
    
    healthChanges.merge(lastValueHeartbeat)
      .flatMap {
        case AppHealth.SetupStatus.AwaitingHealth =>
.....
```

I'm not the best with `Pull`s so could you tell me if this does the same thing? If so, which version do you prefer? :)

Reply:
Your way is very concise, and so yeah it is very tempting to pick yours.

There is a difference.  Imagine the configured heartbeat period is 60 minutes.  Imagine the last heartbeat was at 10:00 am.  Then imagine at 10:55 am the app transitions from healthy to unhealthy.

With your implementation we send an unhealthy alert at 10:55 and at 11:00.   Whereas with my implementation the unhealthy alert happens at 10:55 only (and thereafter at 11:55).

Hmmm..... I need to weigh up benefits of simplicity vs benefits of regularly spaced alerts.

Reply:
I've decided to leave it with the Pulls for now.  Elsewhere we are doing a spike to consume the results of this webhook, and that might raise new requirements for how and when we send the heartbeats when healthy or unhealthy.  So I can revisit this question if new requirements arise.